server/backend/api/app/routers/router_guards.py

```diff
@@ -273,3 +361,78 @@ async def enforce_self_grant_guard(
                 },
                 headers={"WWW-Authenticate": "Bearer"},
             )
+
+
+async def enforce_bceid_by_same_org_guard(
+    request: Request,
+    # forbid business bceid user (requester) manage idir user's access
+    _enforce_user_type_auth: None = Depends(authorize_by_user_type),
+    requester: Requester = Depends(get_current_requester),
+    target_user: Union[TargetUser, None] = Depends(get_target_user_from_id),
+):
+    """
+    When requester is a BCeID user, enforce requester can only manage
+    target user from the same organization.
+    :param _enforce_user_type_auth, call the authorize_by_user_type method to ensure that
+    this enforce_bceid_by_same_org_guard method only checks when business bceid user grants access to another business bceid user
+    """
+    LOGGER.info(
+        f"Verifying requester {requester.user_name} (type {requester.user_type_code}) "
+        "is allowed to manage BCeID target user for the same organization."
+    )
+    if requester.user_type_code == UserType.BCEID:
+        requester_business_guid = requester.business_guid
+        target_user_business_guid = None
+
+        if target_user:
+            # target_user found, expect it to have business_guid
+            target_user_business_guid = target_user.business_guid
+        else:
+            # target_user does not exist in FAM database (new user)
+            # search the target user from IDIM proxy and get business_guid
+            # todo: update to search by user_guid instead of user_name
```

Comment:
It's very minor, "todo" might be better with "TODO" to be tracked by some system I think (like typescript: `[S1135 20](https://rules.sonarsource.com/typescript/RSPEC-1135): Track uses of “TODO” tags`].